grpc-testtool/benches/insert.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use rand::{distributions::Alphanumeric, Rng, SeedableRng};
+use std::{
+    borrow::BorrowMut as _, cell::RefCell, env, fs::remove_dir_all, net::TcpStream,
+    os::unix::process::CommandExt, path::PathBuf, thread::sleep, time::Duration,
+};
+
+use rpc::rpcdb::{self, PutRequest, WriteBatchRequest};
+pub use rpc::service::Database as DatabaseService;
+
+use rpcdb::database_client::DatabaseClient;
+
+use async_once_cell::OnceCell;
+
+use std::process::Command;
+
+/// The directory where the database will be created
+const TESTDIR: &str = "/tmp/benchdb";
+/// The port to use for testing
+const TESTPORT: u16 = 5000;
+/// The URI to connect to; this better match the TESTPORT
+const TESTURI: &str = "http://localhost:5000";
+/// Retry timeouts (in seconds); we want this long for processes
+/// to start and exit
+const RETRY_TIMEOUT_SEC: u32 = 5;
+
+/// Merkledb configuration options; these are ignored by the rust side
+/// These were chosen to be as close to the defaults for firewood as
+/// possible. NodeCacheSize is a guess.
+const MERKLEDB_OPTIONAL_CONFIGURATIONS: &str = r#"
+    {
+    "BranchFactor":16,
+    "ProcessName":"bench",
+    "HistoryLength":100,
+    "NodeCacheSize":1000
+    }
+"#;
+
+/// Clean up anything that might be left from prior runs
+fn reset_everything() {
+    // kill all process servers
+    Command::new("killall")
+        .arg("process-server")
+        .output()
+        .expect("killall should work");
+
+    // wait for them to die
+    retry("process server pids", || process_server_pids().is_empty());
+
+    // remove test directory
+    remove_dir_all(TESTDIR).ok();
+
+    // find the process server
+    let process_server = process_server_path().expect("Can't find process-server on path");
+    eprintln!("Using process-server {}", process_server.display());
+
+    // spawn a new one; use a separate thread to avoid zombies
+    std::thread::spawn(|| {
```

Comment:
It seem we don't kill this at the end of the benchmarks. I feel like we should do that, no?